app/js/script.js

```diff
@@ -0,0 +1,173 @@
+  document.querySelector(
+    "#minute"
+  ).style.transform = `rotate(${minuteHand}deg)`;
+  document.querySelector(
+    "#second"
+  ).style.transform = `rotate(${secondHand}deg)`;
+
+  if (customTime) {
+    customTime.setSeconds(customTime.getSeconds() + 1);
+  }
+}
+
+function startClock() {
+  setCurrentTime();
+  intervalId = setInterval(setCurrentTime, 1000);
+}
+
+function stopClock() {
+  clearInterval(intervalId);
+}
+
+startClock();
+
+const timeZoneSelector = document.createElement("select");
+timeZoneSelector.classList.add("timezone-select");
+timeZones.forEach((zone) => {
+  const option = document.createElement("option");
+  option.value = zone.value;
+  option.textContent = zone.name;
+  timeZoneSelector.appendChild(option);
+});
+
+document.querySelector(".main").appendChild(timeZoneSelector);
+
+timeZoneSelector.addEventListener("change", (e) => {
+  currentTimeZone = e.target.value;
+  setCurrentTime();
+});
+
+const manualControlSelector = document.createElement("select");
+manualControlSelector.classList.add("manual-control-select");
+manualControls.forEach((control) => {
+  const option = document.createElement("option");
+  option.value = control.name;
+  option.textContent = control.name;
+  manualControlSelector.appendChild(option);
+});
+
+document.querySelector(".main").appendChild(manualControlSelector);
+
+manualControlSelector.addEventListener("change", (e) => {
+  manualControlMode = e.target.value;
+
+  if (manualControlMode === "Stop to set a time") {
+    stopClock();
+    isPaused = true;
+  } else if (manualControlMode === "Restart a clock") {
+    isPaused = false;
+    customTime = getClockTime();
+    startClock();
+  }
+});
+
+function getAngle(x, y) {
+  const svg = document.querySelector("svg");
+  const centerX = svg.getBoundingClientRect().left + svg.clientWidth / 2;
+  const centerY = svg.getBoundingClientRect().top + svg.clientHeight / 2;
+  let angle = 90 + (Math.atan2(y - centerY, x - centerX) * 180) / Math.PI;
+  return angle;
+}
+
+const hourHand = document.querySelector("#hour");
+const minuteHand = document.querySelector("#minute");
+const secondHand = document.querySelector("#second");
+
+function addHandListener(hand) {
+  hand.addEventListener("mousedown", (event) => {
+    startDraggingHand(hand, event);
+  });
+}
+
+addHandListener(hourHand);
+addHandListener(minuteHand);
+addHandListener(secondHand);
+
+function startDraggingHand(hand, event) {
+  if (!isPaused) return;
+
+  hand.style.transform = `rotate(${getAngle(event.clientX, event.clientY)}deg)`;
+
+  document.onmousemove = function (event) {
+    hand.style.transform = `rotate(${getAngle(
+      event.clientX,
+      event.clientY
+    )}deg)`;
+  };
+
+  document.onmouseup = () => {
+    stopDragging();
+  };
+}
+
+function stopDragging() {
+  document.onmousemove = null;
+  document.onmouseup = null;
+  customTime = getClockTime();
+  //console.log(`Time is set to: ${customTime}`);
+}
```

Comment:
The function is called before the declaration. It's better to move the definition before the call